relentless/ensemble.py

```diff
@@ -1,176 +1,437 @@
-                'kB': self.kB,
-                'T': self.T,
-                'P': self.P,
-                'V': self.V,
-                'N': self.N,
-                'mu': self.mu,
-                'conjugates': self.conjugates}
-        with open('{}.json'.format(basename),'w') as f:
-            json.dump(data, f, sort_keys=True, indent=4)
-
-        # dump rdfs in separate files
-        for pair in self.rdf:
-            if self.rdf[pair] is not None:
-                i,j = pair
-                np.savetxt('{}.{}.{}.dat'.format(basename,i,j), self.rdf[pair].table, header='r g[{},{}](r)'.format(i,j))
+        """Makes a copy of the ensemble.
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            A copy of the ensemble object.
+
+        """
+        return copy.deepcopy(self)
+
+    def save(self, filename):
+        """Saves the ensemble parameters and the RDF values into a JSON file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = []
+
+        # append thermo data
+        thermo = {'T': self.T,
+                  'P': self.P,
+                  'V': self.V,
+                  'mu':self.mu,
+                  'N': self.N,
+                  'kB': self.kB,
+                  'constant': self.constant} #need constant?
+        data.append(thermo)
+
+        # append rdf data
+        rdf = {pair:self.rdf[pair].table for pair in self.rdf}
+        data.append(rdf)
+
+        # dump data to json file
+        with open('{}.json'.format(filename),'w') as f:
+            json.dump(data, f, indent=4)
 
     @classmethod
-    def load(self, basename='ensemble'):
-        with open('{}.json'.format(basename)) as f:
+    def load(self, filename):
+        """Loads ensemble parameters and RDF values from a JSON file into a new
+        :py:class:`Ensemble` object.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            An ensemble object with parameter values from the JSON files.
+
+        """
+        with open('{}.json'.format(filename)) as f:
             data = json.load(f)
 
-        ens = Ensemble(types=data['types'],
-                       T=data['T'],
-                       P=data['P'],
-                       V=data['V'],
-                       N=data['N'],
-                       mu=data['mu'],
-                       kB=data['kB'],
-                       conjugates=data['conjugates'])
+        thermo = data[0]
+        rdf = data[1]
+
+        #reset fluctuating variables?
+
+        ens = Ensemble(T=thermo['T'],
+                       P=thermo['P'],
+                       V=thermo['V'],
+                       mu=thermo['mu'],
+                       N=thermo['N'],
+                       kB=thermo['kB'])
 
         for pair in ens.rdf:
```

Comment:
Again, will need to be careful about cases where some (or all) of the RDFs are not set. Also, you will likely need to unpack the RDF arrays yourself (they won't have a `table` attribute).